modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/PersistentPageMemoryTableStorage.java

```diff
@@ -374,6 +379,53 @@ private IndexMetaTree createIndexMetaTree(
         }
     }
 
+    /**
+     * Returns new {@link GarbageCollectionTree} instance for partition.
+     *
+     * @param tableView Table configuration.
+     * @param partitionId Partition ID.
+     * @param reuseList Reuse list.
+     * @param pageMemory Persistent page memory instance.
+     * @param meta Partition metadata.
+     * @throws StorageException If failed.
+     */
+    private GarbageCollectionTree createGarbageCollectionTree(
+            TableView tableView,
+            int partitionId,
+            ReuseList reuseList,
+            PersistentPageMemory pageMemory,
+            PartitionMeta meta
+    ) {
+        try {
+            boolean initNew = false;
+
+            if (meta.garbageCollectionTreeMetaPageId() == 0) {
+                long rootPageId = pageMemory.allocatePage(tableView.tableId(), partitionId, FLAG_AUX);
+
+                meta.garbageCollectionTreeMetaPageId(lastCheckpointId(), rootPageId);
+
+                initNew = true;
+            }
+
+            return new GarbageCollectionTree(
+                    tableView.tableId(),
+                    tableView.name(),
+                    partitionId,
+                    dataRegion.pageMemory(),
+                    PageLockListenerNoOp.INSTANCE,
+                    new AtomicLong(),
+                    meta.garbageCollectionTreeMetaPageId(),
+                    reuseList,
+                    initNew
+            );
+        } catch (IgniteInternalCheckedException e) {
```

Comment:
I do not think so.

Reply:
> I do not think so.

Please elaborate

Reply:
Since the tree is initialized in the constructor, so there may be an error there, I don’t see the point in fixing it for now, if you don’t agree, you can create a ticket for it.

Reply:
I'm not talking only about this particular case. I can see this pattern everywhere: we catch an IgniteCheckedException and wrap it into a StorageException. What's the point? Can we throw a StorageException directly? And I'm not asking to fix it in this PR, I'm asking if we should create a ticket and fix that later

Reply:
`org.apache.ignite.internal.storage.StorageException` is in module `ignite-storage-api` and `org.apache.ignite.internal.pagememory.tree.BplusTree` is in module `ignite-page-memory`, we can't just change that.